projects/Mallard/src/components/article/types/article.tsx

```diff
@@ -150,6 +180,20 @@ const Article = ({
     const apiUrl = useApiUrl() || ''
     const { issueId } = useIssueSummary()
 
+    // if webUrl is undefined then we attempt to fetch a url to use for sharing
+    const [shareUrl, setShareUrl] = useState(article.webUrl)
+    // we can only attempt to fetch the url if connected
+    const client = useApolloClient()
+    const data = client.readQuery<{ netInfo: { isConnected: boolean } }>({
+        query: gql('{ netInfo @client { isConnected @client } }'),
+    })
+    const isConnected = data && data.netInfo.isConnected
+    const shareUrlFetchEnabled =
+        !shareUrl &&
+        isConnected &&
+        // TODO: remove remote switch once we are happy this feature is stable
```

Comment:
rm

Reply:
The idea with this is that I've got small concerns this feature could cause problems so have released it behind a switch - but a few releases later we should remove that switch so wanted to make it clear that it's a temporary thing. Maybe I should just stick it in a trello card but this feels useful for future editions archaeologists in case I forget to clean it up!